src/Services/InsuranceService.php

```diff
@@ -60,25 +79,116 @@ public function validate($data)
         return $validator->validate($data);
     }
 
-    public function getOne($pid, $type)
+    public function getOne($uuid, $type = 'primary')
```

Comment:
as above strategy, migrate old getOne() function to getOneByPid() and change accordingly where it is called in the codebase. then you do not need to have hybrid functions that are doing very different things.

Reply:
Added the function getOneByPid(), made the change in codebase receipts_by_method_report.php and InsuranceRestController.php